sdk/python/kfp/compiler/_op_to_template.py

```diff
@@ -193,6 +193,10 @@ def _op_to_template(op: BaseOp):
             for name, path in output_artifact_paths.items()
         ]
 
+        for output_artifact in output_artifacts:
+            if output_artifact['name'] in ['mlpipeline-ui-metadata', 'mlpipeline-metrics']:
+                output_artifact['optional'] = True
```

Comment:
Is this field working with current argo image?

Reply:
Yes =)
I've discovered that Argo is using slightly broken way to parse YAML, so unknown fields are ignored.
Let's see what the tests say.

Reply:
All tests have passed.

Reply:
I don't see the sample test passing for the last commit. Actually, it seems that it didn't run for the last commit. Do you know why?

Reply:
It did. https://prow.k8s.io/pr-history/?org=kubeflow&repo=pipelines&pr=1260
But the unfinished tests are often shown at the rightmost column under the "Unknown" commit label.

Reply:
sg. thanks.

/lgtm

Reply:
// When I wrote the comment that the tests have passed, the sample test has already completed the main part and was deleting the cluster. I guess, when you've looked at it, the Prow still has not detected it as finished.

/approve